src/common/config.cc

```diff
@@ -790,42 +803,60 @@ int md_config_t::_get_val(const char *key, char **buf, int len) const
     const config_option *opt = &config_optionsp[i];
     if (strcmp(opt->name, k.c_str()))
       continue;
-
-    ostringstream oss;
+ 
     switch (opt->type) {
       case OPT_INT:
-        oss << *(int*)opt->conf_ptr(this);
+        return std::move(config_value_t(*(int*)opt->conf_ptr(this)));
```

Comment:
Minor: return value optimization handles this automatically -- no need for extra "std::move"